packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -341,37 +346,115 @@ export async function makeWallet({
       .map(([_id, offer]) => harden(offer));
   }
 
-  const compileOffer = makeOfferCompiler({
-    E,
-    zoe,
-    registry,
-
-    collections: {
-      idToOffer,
-      brandTable,
-      purseToBrand,
-      issuerToIssuerNames,
-      purseMapping,
-    },
-  });
-  async function addOffer(
-    rawOffer,
-    hooks = undefined,
-    requestContext = { origin: 'unknown' },
-  ) {
-    const { id: rawId } = rawOffer;
+  const compileProposal = proposalTemplate => {
+    const {
+      want = {},
+      give = {},
+      exit = { onDemand: null },
+    } = proposalTemplate;
+
+    const purseKeywordRecord = {};
+
+    const compile = amountKeywordRecord => {
+      return Object.fromEntries(
+        Object.entries(amountKeywordRecord).map(
+          ([keyword, { pursePetname, extent }]) => {
+            const purse = getPurse(pursePetname);
+            purseKeywordRecord[keyword] = purse;
+            const brand = purseToBrand.get(purse);
+            const amount = { brand, extent };
+            return [keyword, amount];
+          },
+        ),
+      );
+    };
+
+    const proposal = {
+      want: compile(want),
+      give: compile(give),
+      exit,
+    };
+
+    return { proposal, purseKeywordRecord };
+  };
+
+  const display = value => fillInSlots(dehydrate(harden(value)));
+
+  const displayProposal = proposalTemplate => {
+    const {
+      want = {},
+      give = {},
+      exit = { onDemand: null },
+    } = proposalTemplate;
+    const compile = pursePetnameExtentKeywordRecord => {
+      return Object.fromEntries(
+        Object.entries(pursePetnameExtentKeywordRecord).map(
+          ([keyword, { pursePetname, extent }]) => {
+            const purse = getPurse(pursePetname);
+            const brand = purseToBrand.get(purse);
+            const amount = { brand, extent };
+            return [keyword, { pursePetname, amount: display(amount) }];
+          },
+        ),
+      );
+    };
+    const proposal = {
+      want: compile(want),
+      give: compile(give),
+      exit,
+    };
+    return proposal;
+  };
+
+  const compileOffer = async offer => {
+    const { inviteHandleBoardId } = offer;
+    const { proposal, purseKeywordRecord } = compileProposal(
+      offer.proposalTemplate,
+    );
+
+    // Find invite in wallet and withdraw
+    const defaultInvitePurse = getPurse(ZOE_INVITE_PURSE_PETNAME);
```

Comment:
Does it make sense to bind to this purse earlier (e.g. upon construction of the wallet), to weather against petname changes?

It's fine if that's not worth the effort.

Reply:
Yeah, I like that idea a lot, thanks!